modules/um_callerid.c

```diff
@@ -227,7 +239,34 @@ h_hdl_privmsg_user(void *vdata)
 	data->approved = ERR_TARGUMODEG;
 }
 
+static void
+check_umode_change(void *vdata)
+{
+	hook_data_umode_changed *data = (hook_data_umode_changed *)vdata;
+	bool changed = false;
+	struct Client *source_p = data->client;
+
+	if (!MyClient(source_p))
+		return;
+
+	if (data->oldumodes & UMODE_OPER && !IsOper(source_p))
+		source_p->umodes &= ~user_modes['@'];
+
+	changed = ((data->oldumodes ^ source_p->umodes) & user_modes['@']);
+
+	if (source_p->umodes & user_modes['@'])
```

Comment:
This should be testing for the transition from `-@` to `+@` specifically. Presumably some aspect of `changed` was intended to handle this.

Reply:
I think that's wrong; if you lose the oper:message privilege you should get +@ taken away too